server.js

```diff
@@ -2,34 +2,237 @@ import express from 'express'
 app.use(cors())
 app.use(bodyParser.json())
+app.use((req, res, next) => {
+  if (mongoose.connection.readyState === 1) {
+    next()
+  } else {
+    res.status(503).json({ error: 'Service unavailable' })
+  }
+})
+
+
+
+const Book = mongoose.model('Book', {
+  bookID: Number,
+  title: {
+    type: String
+
+  },
+  authors: String,
+  average_rating: {
+    type: Number,
+    default: 0,
+  },
+  num_pages: {
+    type: Number,
+    default: 0
+  },
+  ratings_count: {
+    type: Number,
+    default: 0
+  },
+  text_reviews_count: {
+    type: Number,
+    default: 0
+  },
+  img_url: {
+    type: String,
+    default: ''
+  }
+})
+
+const User = mongoose.model('User', {
+  username: {
+    type: String,
+    minlength: 3,
+    maxlength: 20
+  },
+  password: {
+    type: String,
+    minlength: 8
+  },
+  ratings: {
+    type: Array,
+    default: []
+  },
+  profilePicture: {
+    type: String,
+    default: ''
+  }
+})
+
+
+const createRandomBook = async (num, titlesArray, authorArray) => {
+  await Book.deleteMany()
+  const arr = Array.apply(null, Array(num))
+
+  arr.forEach((val, index) => {
+    new Book({
+      bookID: index,
+      title: titlesArray[parseInt(titlesArray.length * Math.random())],
+      authors: authorArray[parseInt(authorArray.length * Math.random())],
+      num_pages: parseInt(Math.random() * 2000),
+      ratings_count: parseInt(Math.random() * 10000),
+      average_rating: (Math.random() * 5).toFixed(2)
+
+    }).save()
+  })
+
+
+}
+if (process.env.SEED_RANDOM) {
+  const generateRandomBooks = () => {
+    const bookTitles = []
+    const authors = []
+    data.forEach((book) => bookTitles.push(book.title))
+    data.forEach((book) => {
+      if (!authors.includes(book.authors.split('-')[0])) {
+        authors.push(book.authors.split('-')[0])
+      }
+    })
+    createRandomBook(500, bookTitles, authors)
+  }
+  generateRandomBooks()
+}
+
+// if (process.env.RESET_DB) {
+//   const seedDatabase = async () => {
+//     await Book.deleteMany()
+
+//     data.forEach((book) => {
+//       new Book(book).save()
+//     })
+//   }
+//   seedDatabase()
+// }
+
 
-// Start defining your routes here
 app.get('/', (req, res) => {
-  res.send('Hello world')
+  res.json({
+    homepage: '/books',
+    queryOrder: '?order=highest, lowest, longest, shortest',
+    queryKeyword: '?keyword=an author or title',
+    queryPage: '?page=1,2,3,4...',
+    individualBook: '/books/1,2,3,4...',
+    keywordResultsCanBeOrdered: 'true'
+  })
+})
+
+
+
+app.get('/books', async (req, res) => {
+  const keyword = req.query.keyword
+  const keyRegExp = new RegExp('\\b' + keyword + '\\b', 'i')
+  const order = req.query.order
+  const page = +req.query.page || 1
+  const PAGE_SIZE = 20
+  let mySort = order === 'highest' ? { average_rating: -1 } : order === 'lowest' ? { average_rating: 1 }
+    : order === 'longest' ? { num_pages: -1 } : order === 'shortest' ? { num_pages: 1 } : { bookID: 1 }
```

Comment:
Nested ternary expressions are crazy nerdy and I personally love them but they are difficult to read. Try to use Switch statement when you have so many options on hand: https://www.w3schools.com/js/js_switch.asp